torch/utils/data/datapipes/dataframe/dataframe_wrapper.py

```diff
@@ -1,53 +1,64 @@
-try:
-    import pandas  # type: ignore[import]
-
-    # pandas used only for prototyping, will be shortly replaced with TorchArrow
-    WITH_PANDAS = True
-except ImportError:
-    WITH_PANDAS = False
-
+_pandas = None
+_WITH_PANDAS = None
+
+def _try_import_pandas() -> bool:
+    try:
+        import pandas  # type: ignore[import]
+        global _pandas
+        _pandas = pandas
+        return True
+    except ImportError:
+        return False
+
+
+# pandas used only for prototyping, will be shortly replaced with TorchArrow
+def _with_pandas() -> bool:
+    global _WITH_PANDAS
+    if _WITH_PANDAS is None:
+        _WITH_PANDAS = _try_import_pandas()
+    return _WITH_PANDAS
 
 class PandasWrapper:
     @classmethod
     def create_dataframe(cls, data, columns):
-        if not WITH_PANDAS:
+        if not _with_pandas():
```

Comment:
@robieta do you know if this kind of change would have significant performance implications?

Reply:
It shouldn't. I did a quick benchmark that puts the import check at 130 ns, compared to 100 **us** to create a trivial dataframe. (And 500 ns to take its `len`.)